docs/9_mcp_workflow_engine_extraction.md

```diff
@@ -0,0 +1,2142 @@
+# MCP Workflow Engine Extraction - Technical Design Document
+
+## Executive Summary
+
+This document specifies the extraction of the LangGraph-based workflow engine from the `mobile-native-mcp-server` package into a new reusable `mcp-workflow` package. This refactoring will enable external teams to leverage our deterministic MCP workflow orchestration capabilities for their own MCP server implementations, while maintaining the `mobile-native-mcp-server` as a reference implementation and active consumer of the extracted engine.
+
+### Document Purpose
+
+This Technical Design Document (TDD) provides comprehensive specifications for the extraction and refactoring work. Upon completion of this document and its associated User Story, developers (human or LLM) should have sufficient detail to implement the changes without additional specification.
+
+### Key Deliverables
+
+1. New `@salesforce/magen-mcp-workflow` package containing reusable workflow engine components
+2. Refactored `mobile-native-mcp-server` package consuming the workflow engine as a dependency
+3. Comprehensive test coverage (≥95%) for both packages
+4. Updated documentation reflecting the new architecture
+5. Zero breaking changes for end users of the mobile-native-mcp-server
+
+---
+
+## Consumer Model and Design Philosophy
+
+### Target Audience
+
+The `mcp-workflow` package is designed as **infrastructure for other teams building MCP servers**. Our primary consumers are:
+
+- **Internal teams** building domain-specific MCP servers within the Salesforce ecosystem
+- **External teams** (post-publishing) who want deterministic workflow orchestration for their MCP tool interactions
+- **Any MCP server author** who needs to orchestrate complex, multi-step processes with LLM interaction
+
+### The Workflow Engine as a Framework
+
+`mcp-workflow` is a **framework, not an application**. It provides:
+
+✅ **What We Provide** (Framework Infrastructure):
+
+- Orchestrator tool for managing workflow execution
+- Base classes for MCP tools and workflow nodes
+- State persistence and checkpointing (`.magen/` directory)
```

Comment:
can this be configurable for non `magen` projects? `magen` consumes this package but is a separate product I think

Reply:
But there's a lingering question here: what are we calling "magen"? I left the `.magen/` naming because I was thinking maybe magen is the engine, but maybe magen is the app generation on top of the engine? At which point, what should we call the engine? It could stand to have a proper name, for cases like this.